lib/services/swagger-api-service.js

```diff
@@ -154,66 +154,87 @@ function swaggerFactory(
+                return Promise.try(function() {
+                    return views.get(viewName);
+                })
+                .then(function(view) {
+                    return Promise.map(res.body, function(element) {
+                        return ejs.render(view.contents, _.merge(element, options));
+                    })
                 })
                 .then(function(collection) {
                     return views.render('collection.2.0.json', { collection: collection });
                 });
             } else {
-                render = views.render(viewName, _.merge(res.body, options));
+                return views.render(viewName, _.merge(res.body, options));
             }
-
-            return render.then(function(data) {
-                var renderedObject;
-
-                // Rendered data should be JSON string.
-                try {
-                    renderedObject = JSON.parse(data);
-                } catch(err) {
-                    throw new Errors.ViewRenderError(err.message);
-                }
-
-                return JSON.stringify(renderedObject);
-            })
-            .catch(function(err) {
-                throw new Errors.ViewRenderError(err.message);
-            });
         })
         .then(function(data) {
-            res.status(status).send(data);
+            // determine content-type
+            var body = data;
+            if (typeof(data) === 'object') {
+                res.set('Content-Type', 'application/json');
+            } else if (typeof(data) === 'string') {
+                body = Promise.try(function() {
+                    var parsed = JSON.parse(data);
+                    res.set('Content-Type', 'application/json');
+                    return JSON.stringify(parsed);
+                })
+                .catch(function(err) {
+                    if (!viewName) {
+                        res.set('Content-Type', 'text/plain');
+                        return data;
+                    } else {
+                        throw new Errors.ViewRenderError(err.message);
+                    }
+                });
+            }
+            return body;
         })
         .catch(function(err) {
+            throw new Errors.ViewRenderError(err.message);
+        });
+    }
+
+    function swaggerRenderer(req, res, viewName, next) {
+        return Promise.try(function() {
+            assert.ok(!res.headersSent, 'headers have already been sent');
+            return [ viewName, req, res ];
+        })
+        .spread(_render)
+        .tap(function() {
+            // Set appropriate HTTP status.
+            if (_.isEmpty(res.body) && req.swagger.options.send204OnEmpty) {
+                res.status(204);
+            } else {
+                res.status(res.locals.errorStatus || req.swagger.options.success || 200);
+            }
+        })
+        .then(res.send.bind(res))
+        .catch(function(err) {
+            console.log(err.stack);
```

Comment:
This is just debug code that I forgot to remove before submitting the PR.  I'll delete it.